### PR TITLE
Vmware notification and event api update

### PR DESCRIPTION
**Pull request by ritika-msft**
Fixes #

## Proposed Changes

  -
  -
  -

### CODE REVIEW

File: DataConnectors/VMware Carbon Black/VMwareCarbonBlack_API_FunctionApp.json [hunk 1]
```diff
@@ -93,7 +93,11 @@
...
+                "description": "Carbon Black API and/or SIEM Level API Key(s) are required. See the documentation to learn more about the [Carbon Black API](https://developer.carbonblack.com/reference/carbon-black-cloud/cb-defense/latest/rest-api/).\n - A Carbon Black **API** access level API ID and Key is required for [Audit](https://developer.carbonblack.com/reference/carbon-black-cloud/cb-defense/latest/rest-api/#audit-log-events) and [Event](https://developer.carbonblack.com/reference/carbon-black-cloud/platform/latest/data-forwarder-config-api/) logs. \n - A Carbon Black **SIEM** access level  API ID and Key is required for [Notification](https://developer.carbonblack.com/reference/carbon-black-cloud/cb-defense/latest/rest-api/#notifications) events."
```
[the remark this answers is not in the document]
Comment (ritika-msft) [reply DataConnectors/VMware Carbon Black/VMwareCarbonBlack_API_FunctionApp.json:96]: As per the latest code, we are using CB API to pull Audit logs, S3 bucket logs for events and Notification.

 For Notifications if no SIEM key/API is provided, in that case S3 bucket information is used. But for users who provide SIEM key and API as they may not have have a presence in AWS S3, **integrationServices/v3/notification** API is being used. This API is not deprecated as could be seen [here](https://developer.carbonblack.com/reference/carbon-black-cloud/authentication/#creating-an-api-key).
 
  
image

[the remark this answers is not in the document]
Comment (ritika-msft) [reply DataConnectors/VMware Carbon Black/VMwareCarbonBlack_API_FunctionApp.json:96]: Also the code implementation is as per the notification to alert migration guide. We can discuss this in detail in the scheduled call on Wednesday, 15-12-21.
image

**Review by rfortress-cb — DISMISSED** [review]
After talking to Ruchita and Sapan, these changes make sense.

Thank you!

**Comment by ritika-msft** [comment]
Attaching few screenshots for UI changes, you can refer these to get an understanding about the look and feel of UI. 

1.	Field “Log Types” is a drop down, user can select which kind of logs they would like to push to workspace.
image

2.	There are 3 sections : API Credentials, SIEM API Credentials and AWS S3 Bucket Credentials.

2a. If the user has selected “Audit” as one of the Log Types. API Credentials section will appear, and the fields will be made mandatory, if “Audit” is not selected, this section will not appear on UI.
image

2b. If the user has selected “Event” as one of the Log Types. AWS S3 Bucket Credentials section will appear and the fields will be made mandatory, if “Event” is not selected, this section will not appear on UI.
image

2c. For “Alerts”, user should select either “Alert supported with SIEM API Credential” or “Alert supported with AWS S3 bucket credentials”, based on whether they have S3 bucket credentials or SIEM credentials.

i.	If the user has selected “Alert supported with SIEM API Credential” as one of the Log Types. SIEM API Credential section will appear and the fields will be made mandatory, otherwise this section will not appear on UI.
image

 
ii.	If the user has selected “Alert supported with AWS S3 bucket credentials” as one of the Log Types. AWS S3 bucket credentials section will appear, and the fields will be made mandatory, otherwise this section will not appear on UI.
image

 
  2d. If all the log types are selected, all 3 sections will appear on UI.
image

**Comment by bobsyourmom** [comment]
so there is currently no supported CarbonBlack connector for Sentinel? 
i.e. the connector in the Sentinel Content Hub is obsolete?

**Pull request merged**